### PR TITLE
Fixing semver parsing when "=" is in the toml version requirement 

### PR DESCRIPTION
**Pull request by deg4uss3r**
Fixes a bug found where "=" begins the version requirement in the `Cargo.toml`

### CODE REVIEW

File: src/cargo_ops/temp_project.rs [hunk 1]
```diff
@@ -653,7 +653,8 @@ fn valid_latest_version(requirement: &str, version: &Version) -> bool {
         (false, false) | (true, false) => true,
         // both are unstable, must be in the same channel
         (true, true) => {
-            let requirement_version = Version::parse(requirement).expect("Error could not parse requirement into a semantic version"); 
+            let requirement_mod = requirement.replace("=", "").replace(" ", "");
```
Comment (kornelski) [inline src/cargo_ops/temp_project.rs]: This could be simplified to `requirement.trim_left_matches(&['=',' '][..])`

Comment (deg4uss3r) [reply src/cargo_ops/temp_project.rs]: @kornelski I think I want to replace the line with `requirement = requirement.trim_start_matches(&['=',' ','~','^'][..]);
` to allow for more replacements of the toml version specification options. 

Can you think of other scenarios, or edge cases that might case a panic parsing the semantic version here? I tried `*` but that gets resolved before the version is sent to this function, so that passes the tests.

Comment (kornelski) [reply src/cargo_ops/temp_project.rs]: Yeah I've meant `start`.

Do versions like `1.*` appear here? Or `1 || 2`?
